### PR TITLE
add switch to wifi

### PR DESCRIPTION
**Pull request by moinmoin-sh**
added compile time switch to change to wifi instead of ethernet

as discussed. please have a look

### CODE REVIEW

File: src/OXRS_Rack32.cpp [hunk 1]
```diff
@@ -443,6 +486,30 @@ void OXRS_Rack32::_initialiseEthernet(byte * mac)
...
+#else
+void OXRS_Rack32::_initialiseWifi(byte * mac)
+{
+ // Get ESP32 base MAC address
+  WiFi.macAddress(mac);
+  
+  // Display the MAC address on serial
+  char mac_display[18];
+  sprintf_P(mac_display, PSTR("%02X:%02X:%02X:%02X:%02X:%02X"), mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);
+  Serial.print(F("[ra32] wifi mac address: "));
```
Comment (sumnerboy12) [inline src/OXRS_Rack32.cpp:498]: Maybe add `ethernet` to the logging on lines 463 and 479, for consistency?

**Review by sumnerboy12 — COMMENTED** [review]
Just thinking about this - do we want to only support one or the other? It might be useful to have both enabled at the same time, for failover? But that would require two `_mqtt` clients, two `_api` clients, etc etc.

Starts getting overly complex I guess. 

I think what you have done is fine. We just need to add that captive portal to remove the need for custom config and compiling. 

Good to have a branch with working code tho - great stuff Moin!

File: src/OXRS_Rack32.cpp (hunk 1 above)
Comment (moinmoin-sh) [reply src/OXRS_Rack32.cpp:498]: I had the same thought regarding enable both ast the same time for failover or runtime switchable option but didn't go further into that due to the coplexity. We can leave this branch until we have time to add the captive portal.

**Comment by sumnerboy12** [comment]
Forgot about this branch!! Just merged #48 which is almost line-for-line the same as this - whoops! My change uses WiFiManager to manage provisioning of AP creds tho - so no need for hardcoded creds.

Closing this PR.

**Pull request closed without merging**